src/services/UserService.ts

```diff
@@ -112,7 +115,22 @@ export class UserService {
 		return user.toJSONPrivate();
 	}
 
-	public async putUserProfile(id: string, options: ProfileDataToCreate, file?: Express.Multer.File) {
+	public async reportUser(id: string, user: User, options: APIReport) {
+		return getConnection().transaction(async entityManager => {
+			const report = new Report();
+			report.id = v4();
+			const { currentTime, description } = options;
+			Object.assign(report, { currentTime, description });
+			report.reportedUser = user;
+			report.reportingUser = await entityManager.findOneOrFail(User, { id });
+			user.report = report;
+			await entityManager.save(user).catch(() => Promise.reject(new APIError(HttpCode.BadRequest, PutProfileError.InvalidEntryDetails)));
```

Comment:
```suggestion
			await entityManager.save(report).catch(() => Promise.reject(new APIError(HttpCode.BadRequest, PutProfileError.InvalidEntryDetails)));
```

Instead of updating user (which is less efficient) we can just save the report directly.